### DISCUSSION TITLE
FORCE HTTPS = true causes infinite 301 redirect

### DISCUSSION THREAD

**Discussion by redlewel** [outline/outline#2598] [Self Hosting] [answered]
So when I have the environment variable FORCE HTTPS = true in my .env, I get an infinite 301 redirect to the site: https://imgur.com/bQwoSXL.png
When the FORCE HTTPS is turned off the .env file the site works and loads normally but I am unable to upload images and I see that it is directing to the http address of the docker container in network inspector:
https://imgur.com/sQqrIqB.png
Do I need FORCE_HTTPS set to true in order to have ssl with images?  I modified my nginx configuration file for the site multiple times.  I have outline configured in a docker container and nginx on the host OS proxing to the docker container IP.  Is there some type of ssl redirect I need to implement to have the site work properly?

**Comment by tommoor** [comment]
You probably need to include `X-Forwarded-For` header if you're terminating https before it's reaching Outline, so that the app understands it _is_ on https and doesn't redirect.

**Reply by redlewel** [reply]
I already have that line set in my nginx configuration I can show you here:

Are you suggesting that I include the `X-Forwarded-For` with `FORCE HTTPS` set to `true` or `false`?

^^^^ Irrelevant, problem did not have anything to do with nginx

**Comment by redlewel** [comment, marked as answer]
I figured it out.  The problem had nothing to do with HTTPS redirection or the `FORCE_HTTPS` .env variable.  MinIO wasn't working!  That was the http address it was redirecting to.  I tried for awhile to get MinIO working with outline, but unfortunately I found no way to modify CORS within MinIO.  

I decided to spin up an Amazon S3 bucket and now the site works!  Now the entire site works as I want and the lock is present on the address bar.
